### PR TITLE
Updated profile endpoint to v5

### PR DESCRIPTION
**Pull request by majsavli**

### CODE REVIEW

File: brd-android/rockwallet-exchange/src/main/java/com/rockwallet/exchange/ui/features/buy/input/BuyInputContract.kt [hunk 1]
```diff
@@ -219,7 +219,7 @@ interface BuyInputContract {
                 }
 
             val isAchAvailable: Boolean
-                get() = profile?.accessRights?.achAccess == true
+                get() = profile?.accessRights?.buyAch?.hasAccess == true
```
Comment (rok-n) [inline brd-android/rockwallet-exchange/src/main/java/com/rockwallet/exchange/ui/features/buy/input/BuyInputContract.kt:222]:
```suggestion
                get() = profile?.isAchAllowed()
```

File: brd-android/kyc/src/main/java/com/rockwallet/kyc/ui/features/status/VerificationStatusViewModel.kt [hunk 2]
```diff
@@ -28,7 +28,7 @@ class VerificationStatusViewModel(
         profileManager.updateProfile()
 
         setState {
-            copy(isAchAvailable = profileManager.getProfile()?.accessRights?.achAccess ?: false)
+            copy(isAchAvailable = profileManager.getProfile()?.accessRights?.buyAch?.hasAccess ?: false)
```
Comment (rok-n) [inline brd-android/kyc/src/main/java/com/rockwallet/kyc/ui/features/status/VerificationStatusViewModel.kt:31]:
```suggestion
            copy(isAchAvailable = profileManager.getProfile().isAchAllowed())
```

File: brd-android/app/src/main/java/com/breadwallet/ui/navigation/RouterNavigator.kt [hunk 3]
```diff
@@ -338,7 +338,7 @@ class RouterNavigator(
                     SsnStatus.UNKNOWN -> throw IllegalStateException("This should not be reached")
                     SsnStatus.FULL -> ExchangeActivity.StartDestinations.Sell.Default(
                         targetWithdrawalCryptoCurrency = effect.targetWithdrawalCryptoCurrency,
-                        screenType = when (profile.accessRights?.achSellAccess) {
+                        screenType = when (profile.accessRights?.sellAch?.hasAccess) {
                             false -> SellScreenType.WITHDRAW_CARD
                             else -> SellScreenType.WITHDRAW_ACH
                         }
```
Comment (rok-n) [inline brd-android/app/src/main/java/com/breadwallet/ui/navigation/RouterNavigator.kt:344]:
```suggestion
                        screenType = if (profile.isSellAchAllowed()) { SellScreenType.WITHDRAW_ACH } else { SellScreenType.WITHDRAW_CARD }
```

File: brd-android/app/src/main/java/com/breadwallet/util/ExchangeAccessRestrictionManager.kt [hunk 4]
```diff
@@ -78,6 +77,18 @@ class ExchangeAccessRestrictionManager(
...
 
+    private fun getRestrictionReason(
+        profile: Profile?,
+        feature: FeatureUnavailableController.Feature
+    ): AccessRestrictionReason? {
+        return when (feature) {
+            FeatureUnavailableController.Feature.BUY_CARD -> profile?.accessRights?.buyCard?.restrictionReason
+            FeatureUnavailableController.Feature.BUY_ACH -> profile?.accessRights?.buyAch?.restrictionReason
+            FeatureUnavailableController.Feature.SWAP -> profile?.accessRights?.swap?.restrictionReason
+            FeatureUnavailableController.Feature.SELL -> profile?.accessRights?.sellAch?.restrictionReason
```
Comment (rok-n) [inline brd-android/app/src/main/java/com/breadwallet/util/ExchangeAccessRestrictionManager.kt:88]: We should show the restriction reason for Sell Card too. I know that this feature is currently disabled, but the user can have an access to SELL CARD only (non US residents) in the future.

File: brd-android/rockwallet-network/src/main/java/com/rockwallet/network/registration/model/Profile.kt [hunk 5]
```diff
@@ -57,15 +57,13 @@ fun Profile?.isNotVerified() = this?.kycStatus == KycStatus.EMAIL_VERIFICATION_P
 
 fun Profile?.isCurrentStatusEmailVerified() = this?.kycStatus == KycStatus.EMAIL_VERIFIED
 
-fun Profile?.isBuyAllowed() = this?.accessRights?.buyAccess == true
+fun Profile?.isBuyAllowed() = this?.accessRights?.buyCard?.hasAccess == true
```
Comment (rok-n) [inline brd-android/rockwallet-network/src/main/java/com/rockwallet/network/registration/model/Profile.kt:60]:
```suggestion
fun Profile?.isBuyCardAllowed() = this?.accessRights?.buyCard?.hasAccess == true
```

File: brd-android/rockwallet-network/src/main/java/com/rockwallet/network/registration/model/Profile.kt [hunk 6]
```diff
@@ -57,15 +57,13 @@ fun Profile?.isNotVerified() = this?.kycStatus == KycStatus.EMAIL_VERIFICATION_P
...
 
-fun Profile?.isAchAllowed() = this?.accessRights?.achAccess == true
+fun Profile?.isAchAllowed() = this?.accessRights?.buyAch?.hasAccess == true
```
Comment (rok-n) [inline brd-android/rockwallet-network/src/main/java/com/rockwallet/network/registration/model/Profile.kt:62]:
```suggestion
fun Profile?.isBuyAchAllowed() = this?.accessRights?.buyAch?.hasAccess == true
```

File: brd-android/rockwallet-exchange/src/main/java/com/rockwallet/exchange/ui/features/sell/ssn/SellSsnInputViewModel.kt [hunk 7]
```diff
@@ -55,7 +55,7 @@ class SellSsnInputViewModel(
                         setEffect {
                             SellSsnInputContract.Effect.GoToSellInput(
                                 targetWithdrawalCryptoCurrency = arguments.targetWithdrawalCryptoCurrency,
-                                screenType = when (profile?.accessRights?.achSellAccess) {
+                                screenType = when (profile?.accessRights?.sellAch?.hasAccess) {
                                     false -> SellScreenType.WITHDRAW_CARD
                                     else -> SellScreenType.WITHDRAW_ACH
                                 }
```
Comment (rok-n) [inline brd-android/rockwallet-exchange/src/main/java/com/rockwallet/exchange/ui/features/sell/ssn/SellSsnInputViewModel.kt:61]:
```suggestion
                                screenType = if (profile.isSellAchAllowed()) {
                                    SellScreenType.WITHDRAW_ACH
                                } else {
                                    SellScreenType.WITHDRAW_CARD
                                }
```

File: brd-android/app/src/main/java/com/breadwallet/util/ExchangeAccessRestrictionManager.kt (hunk 4 above)
Comment (majsavli) [reply brd-android/app/src/main/java/com/breadwallet/util/ExchangeAccessRestrictionManager.kt:88]: I will do this in a separate MR

**Review by rok-n — APPROVED** [review]

**Pull request merged**